optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/bi/BiConstraintStream.java

```diff
@@ -209,19 +209,36 @@
      */
     <GroupKey_> UniConstraintStream<GroupKey_> groupBy(BiFunction<A, B, GroupKey_> groupKeyMapping);
 
-    /*
-    // TODO implement this
+    /**
+     * Convert the {@link BiConstraintStream} to a different {@link BiConstraintStream}, consisting of unique tuples.
+     * <p>
+     * The first fact is the value resulting from applying the group key mapping function on the original tuple.
+     * The second fact is the value of applying a given {@link BiConstraintCollector} on all the tuples with the same
```

Comment:
"the value" is too broad. 
Can we do something like:
- The first fact is the return value of the first group key mapping function, applied on the incoming tuple.
- The second fact is the result value of ...